R/compare.R

```diff
@@ -1,19 +1,19 @@
 #' Comparison proxy
 #'
 #' Returns a proxy object (i.e. an atomic vector or data frame of atomic
-#' vectors). For [vctr]s, this determines the behaviour of [order()] and
-#' [sort()] (via [xtfrm()]); `<`, `>`, `>=` and `<=` (via [vec_compare()]);
-#' and [min()], [max()], [median()], and [quantile()].
+#' vectors). For [vctr]s, this determines the behaviour of `<`, `>`, `>=`
+#' and `<=` (via [vec_compare()]); and [min()], [max()], [median()], and
+#' [quantile()].
 #'
 #' The default method assumes that all classes built on top of atomic
-#' vectors or records are orderable. If your class is not, you will need
+#' vectors or records are comparable. Internally the default calls
+#' [vec_proxy_equal()]. If your class is not comparable, you will need
 #' to provide a `vec_proxy_compare()` method that throws an error.
 #'
+#' Lists are not comparable, as it is unclear how to compare elements of
```

Comment:
"unclear" makes it sound like we don't know how to do it but it might be possible in the future. Perhaps:

> as comparing elements of different types is undefined.

Reply:
Though now I wonder if we should transform lists to a `list_of()` (at least conceptually). This would fail if no common type, and then we can compare the common prefix of the list elements, e.g. `list(c(1, 2, 4))` is lesser than `list(1:10)`. We'd have a rule in case of equal prefixes.

Reply:
I think the general concept is that comparison requires a common type but ordering doesn't. Maybe add this rule to the description?

Reply:
I wouldn't want to claim such a general principle; I think we should just state that lists have special behaviour and describe what it is.